### PR TITLE
Sieve of Eratosthenes optimization.

### PR DESCRIPTION
**Pull request by izanbf1803**
Complexity and space reduced to a half.

### CODE REVIEW

**Comment by ankit167** [comment]
The code gives wrong output for input- 0 and 1

File: math/primes_sieve_of_eratosthenes.py [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
-    sieve = [1 for v in range(x+1)]     # Sieve
-    primes = []                         # List of Primes
-    for i in range(2,x+1):
-        if sieve[i]==1:
-            primes.append(i)
-            for j in range(i,x+1,i):
+    assert(x >= 0)
+    # If x is even, exclude x from list (-1):
+    sieve_size = (x//2 - 1) if x % 2 == 0 else (x//2)
+    sieve = [1 for v in range(sieve_size)] # Sieve
```
Comment (ankit167) [inline math/primes_sieve_of_eratosthenes.py]: Atleast two spaces before inline comment.

File: math/primes_sieve_of_eratosthenes.py [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
+    primes = [2] # List of Primes
```
Comment (ankit167) [inline math/primes_sieve_of_eratosthenes.py]: Atleast two spaces before inline comment.

File: math/primes_sieve_of_eratosthenes.py [hunk 3]
```diff
@@ -1,14 +1,38 @@
...
+    for i in range(0, sieve_size):
+        if sieve[i] == 1:
+            value_at_i = i*2 + 3
+            primes.append(value_at_i)
+            for j in range(i, sieve_size, value_at_i):
                 sieve[j]=0
```
Comment (ankit167) [inline math/primes_sieve_of_eratosthenes.py]: missing white spaces around operator

**Comment by izanbf1803** [comment]
Fixed error for input, spaces before comment and operator.

**Pull request merged**